File: Serene3/Serene3.Web/Modules/com/cities/citiesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Serene3.com.Columns;

[ColumnsScript("com.cities")]
[BasedOnRow(typeof(citiesRow), CheckNames = true)]
public class citiesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string StateName { get; set; }
}

File: Serene3/Serene3.Web/Modules/ServerTypes/com/citiesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { citiesRow } from "./citiesRow";

export interface citiesColumns {
    Id: Column<citiesRow>;
    Name: Column<citiesRow>;
    StateName: Column<citiesRow>;
}

export class citiesColumns extends ColumnsBase<citiesRow> {
    static readonly columnsKey = 'com.cities';
    static readonly Fields = fieldsProxy<citiesColumns>();
}